pkg/kstatus/observe/aggregator/aggregator.go

```diff
@@ -0,0 +1,62 @@
+package aggregator
+
+import (
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/common"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/wait"
+)
+
+// AllCurrentOrNotFoundStatusAggregator is an implementation of the StatusAggregator interface
+// that considers the aggregate status as Current whenever all resources have either the Current
+// or the NotFound status.
```

Comment:
This is another "corner case" where I haven't fully figured out what is the best solution. NotFound is treated as Current here to handle situations with deletions gracefully. When we have an apply function that includes pruning or a teardown/destroy command, we will consider things fully reconciled when all resources have been deleted. But I agree, it seems counterintuitive that NotFound is considered Current.

Reply:
> When we have an apply function that includes pruning or a teardown/destroy command

That's an interesting case indeed. Krane actually ignores the torn down resources (other than printing a claim that they were pruned) for the purposes of convergence determination. My assumption is that the service working as intended doesn't generally depend on shutdown, and if it does, that will surface other ways (e.g. in Recreate deployments). But if we can do better here, that would be great. Maybe this lib needs to know which resources it expects to converge vs disappear? Because if a resource you asked to watch is not there, I'm not so sure about assuming it was supposed to be removed.